src/niscope/system_tests/test_system_niscope.py

```diff
@@ -1,527 +1,588 @@
-
-
-# Basic usability tests
-def test_read(multi_instrument_session):
-    test_voltage = 1.0
-    test_record_length = 2000
-    test_num_channels = 2
-    test_num_records = 3
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
-    waveforms = multi_instrument_session.channels[test_channels].read(num_samples=test_record_length, num_records=test_num_records)
-    assert len(waveforms) == test_num_channels * test_num_records
-    for i in range(len(waveforms)):
-        assert len(waveforms[i].samples) == test_record_length
-
-
-def test_fetch(multi_instrument_session):
-    test_voltage = 1.0
-    test_record_length = 2000
-    test_num_channels = 2
-    test_starting_record_number = 2
-    test_num_records_to_acquire = 5
-    test_num_records_to_fetch = test_num_records_to_acquire - test_starting_record_number
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records_to_acquire, True)
-    with multi_instrument_session.initiate():
-        waveforms = multi_instrument_session.channels[test_channels].fetch(
-            num_samples=test_record_length,
-            record_number=test_starting_record_number,
-            num_records=test_num_records_to_fetch)
-    assert len(waveforms) == test_num_channels * test_num_records_to_fetch
-    expected_channels = test_channels.split(',') * test_num_records_to_fetch
-    expected_records = [2, 2, 3, 3, 4, 4]
-    for i in range(len(waveforms)):
-        assert len(waveforms[i].samples) == test_record_length
-        assert waveforms[i].channel == expected_channels[i]
-        assert waveforms[i].record == expected_records[i]
-
-
-def test_fetch_defaults(multi_instrument_session):
-    test_voltage = 1.0
-    test_record_length = 2000
-    test_num_channels = 2
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, 1, True)
-    with multi_instrument_session.initiate():
-        waveforms = multi_instrument_session.channels[test_channels].fetch()
-    assert len(waveforms) == test_num_channels
-    for i in range(len(waveforms)):
-        assert len(waveforms[i].samples) == test_record_length
-
-
-@pytest.fixture(params=[(1000, 1000), (2000, 2000), (3000, 2000)], ids=["less_than_actual", "equal_to_actual", "greater_than_actual"])
-def measurement_wfm_length(request):
-    MeasWfmLength = collections.namedtuple('MeasurementWaveformLength', ['passed_in', 'expected'])
-    return MeasWfmLength(passed_in=request.param[0], expected=request.param[1])
-
-
-def test_fetch_array_measurement(multi_instrument_session, measurement_wfm_length):
```

Comment:
`test_fetch_array_measurement` and `test_fetch_array_measurement_defaults` are just in the SystemTests class below. There's a TODO above them to support them on grpc but right now they're broken so they're only tested against the `_library_interpreter` for now.